PowerEditor/src/functionList.xml

```diff
@@ -88,21 +88,21 @@
 						<nameExpr expr="\w+\Z" />
 					</className>
 					<function
-						mainExpr="^\s*((static|public|protected|private|final)*(\s+(static|public|protected|private|final))+\s+)?(function\s+)+(\w+(\s+\w+)?(\s+|\*\s+|\s+\*|\s+\*\s+))?(\w+\s*::)?(?!(if|while|for|switch))[\w~]+\s*\([^\{]*\{"
```

Comment:
Negative look-a-head is NOT superfluous!
It should have been `(?!(?-i:if|while|for|switch)\b)` to ...
1. prevent (i.e. the `(?! ... )`) the case-sensitive keywords (i.e. the `(?-i: ... )`) from being used as identifiers;
2. permit identifiers to _start_ with a keyword by adding the word boundary (`\b`).